src/traces/box/defaults.js

```diff
@@ -137,25 +138,50 @@ function handleSampleDefaults(traceIn, traceOut, coerce, layout) {
                 break;
             case '02':
                 defaultOrientation = 'v';
-                len = Math.min(sLen, yLen);
+                len = Math.min(sLen, y.length);
                 break;
             // both
             case '12':
                 defaultOrientation = 'v';
-                len = Math.min(sLen, xLen, yLen);
+                len = Math.min(sLen, xLen, y.length);
                 break;
             case '21':
                 defaultOrientation = 'h';
-                len = Math.min(sLen, xLen, yLen);
+                len = Math.min(sLen, x.length, yLen);
                 break;
             case '11':
                 // this one is ill-defined
                 len = 0;
                 break;
             case '22':
-                // this one case happen on multi-category axes
-                defaultOrientation = 'v';
-                len = Math.min(sLen, xLen, yLen);
+                var hasCategories = false;
+                var i;
```

Comment:
Non-blocking: IMO it would be nice to declare these two vars outside the switch block.

Reply:
I'll pass if ok.